flink-connectors/flink-connector-kafka-base/src/main/java/org/apache/flink/streaming/connectors/kafka/KafkaTableSinkBase.java

```diff
@@ -57,12 +55,11 @@
 	protected final Properties properties;
 
 	/** Serialization schema for encoding records to Kafka. */
-	protected Optional<SerializationSchema<Row>> serializationSchema;
+	protected final Optional<SerializationSchema<Row>> serializationSchema;
 
 	/** Partitioner to select Kafka partition for each item. */
 	protected final Optional<FlinkKafkaPartitioner<Row>> partitioner;
 
-	// legacy variables
 	protected String[] fieldNames;
```

Comment:
I think we also don't need these variables anymore.

Reply:
Done.